### PR TITLE
New distributed isomorphism integration tests

### PR DESCRIPTION
**Pull request by jessberg**
For every previous centralized query test for the simulator, there is now a distributed test as well.

### CODE REVIEW

File: test_filter_outputs/generic_cargo.toml [hunk 1]
```diff
@@ -19,5 +19,6 @@ rpc_lib = { path = "../../../tracing_sim/libs/rpc_lib" }
 graph_utils = { path = "../../../tracing_sim/libs/graph_utils" }
 serde = { version = "1.0", features = ["derive"] }
 indexmap = { version = "1.6.1", features = ["serde-1"] }
+serde_yaml = "0.8"
```
Comment (fruffy) [inline test_filter_outputs/generic_cargo.toml:22]: What do we need this for? I do not remember having to serialize yaml files.

Comment (jessberg) [reply test_filter_outputs/generic_cargo.toml:22]: So serde_json does not put in empty entries.  But the isomorphism algorithm assumes those empty entries exist, they are just empty.  We could make our own serializer/deserializer for JSON, but it was easier to just to switch to using YAML, which by default does contain the entries even if they are empty.

Comment (fruffy) [reply test_filter_outputs/generic_cargo.toml:22]: Okay can we remove serde_json then?

Comment (jessberg) [reply test_filter_outputs/generic_cargo.toml:22]: Well, we're still using JSON for the regular, centralized filter.  I can make another pull request to make that yaml instead though.

Comment (fruffy) [reply test_filter_outputs/generic_cargo.toml:22]: Yeah that would be good, does not make sense to have multiple formats

**Review by fruffy — APPROVED** [review]

**Pull request merged**